libraries/redshop/helper/shipping.php

```diff
@@ -848,7 +848,7 @@ public static function listShippingRates($shippingClass, $usersInfoId, &$data)
 				AND (( " . $db->quote($weightTotal) . " BETWEEN " . $db->qn('shipping_rate_weight_start')
                 . " AND " . $db->qn('shipping_rate_weight_end') . ") OR ("
                 . $db->qn('shipping_rate_weight_end') . " = 0)) " . $whereShippingVolume . "
-				AND (" . $db->qn('shipping_rate_on_product') . " = '' " . $pWhere . ") AND ("
+				AND (" . $db->qn('shipping_rate_on_product') . " = '' " . $pWhere . ") OR ("
                 . $db->qn('shipping_rate_on_category') . " = '' " . $cWhere . ")" . $where . "
```

Comment:
`AND (" . $db->qn('shipping_rate_on_product') . " = '' " . $pWhere . ") OR ("
                . $db->qn('shipping_rate_on_category') . " = '' " . $cWhere . ")"`

I don't think this part is correct, the format of `$pWhere` is `OR (....)` , so that the whole where logic should be the composition of  whole where of product selection and whole where of category selection separated by OR in two group.